src/ui/screens/ProposeTipScreen.tsx

```diff
@@ -11,14 +11,32 @@ import {Padder} from '@ui/components/Padder';
 import {Account} from 'src/api/hooks/useAccount';
 import {useRefetch} from 'src/api/hooks/useRefetch';
 import {TIPS_QUERY} from 'src/api/hooks/useTips';
+import {InputLabel} from '@ui/library/InputLabel';
+import {isAddressValid} from 'src/utils/address';
+import {NetworkContext} from 'context/NetworkContext';
 
 export function ProposeTipScreen({navigation}: {navigation: NavigationProp<DashboardStackParamList>}) {
-  const {colors} = useTheme();
   const [state, dispatch] = useReducer(reducer, initialState);
   const startTx = useApiTx();
   const {refetch: refetchTips} = useRefetch([TIPS_QUERY]);
+  const {currentNetwork} = useContext(NetworkContext);
 
-  const valid = state.account && state.beneficiary && state.reason && state.reason.length > 4;
+  const isBeneficiaryAddressValid = useMemo(() => {
+    const payload = state.beneficiary
+      .replace(/\r?\n|\r/g, ' ')
+      .split(' ')
+      .join('');
+    dispatch({type: 'SET_BENEFICIARY', payload});
+    return state.beneficiary ? isAddressValid(currentNetwork, state.beneficiary) : false;
+  }, [state.beneficiary, currentNetwork]);
+
+  const isTipReasonValid = useMemo(() => {
+    const payload = state.reason.replace(/\r?\n|\r/g, ' ');
+    dispatch({type: 'SET_REASON', payload});
```

Comment:
Can't imagine a use case that requires a reducer dispatch from a `useMemo`.
There's a risk of falling into an infinite loop.

Possibly you don't need the `useMemo` here. What is the purpose of this `state.reason.replace(/\r?\n|\r/g, ' ')` regex here?